From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Debayan Kar <kardebayan3@gmail.com>
Date: Sun, 25 Oct 2026 16:03:35 +0000
Subject: [PATCH 07/12] Skip MIUI SDK initialization

miui.external.Application loads and starts the MIUI SDK from
com.miui.core before the app runs, and replaces the instrumentation
with an "SDK missing" error screen when it is not there. Everything
the app needs from the SDK is now bundled, so report it as ready.
---
 .../miui/external/Application.smali           | 192 +-----------------
 1 file changed, 7 insertions(+), 185 deletions(-)

diff --git a/smali_classes2/miui/external/Application.smali b/smali_classes2/miui/external/Application.smali
index 7cb9c820..720ccf38 100644
--- a/smali_classes2/miui/external/Application.smali
+++ b/smali_classes2/miui/external/Application.smali
@@ -132,205 +132,27 @@
 .end method
 
 .method private initializeSdk()Z
-    .locals 8
-
-    const-string v0, "initialize"
-
-    const/4 v1, 0x0
-
-    :try_start_0
-    new-instance v2, Ljava/util/HashMap;
-
-    invoke-direct {v2}, Ljava/util/HashMap;-><init>()V
-
-    invoke-static {}, Lmiui/external/SdkEntranceHelper;->getSdkEntrance()Ljava/lang/Class;
-
-    move-result-object v3
-
-    const/4 v4, 0x2
-
-    new-array v5, v4, [Ljava/lang/Class;
-
-    const-class v6, Landroid/app/Application;
-
-    aput-object v6, v5, v1
-
-    const-class v6, Ljava/util/Map;
-
-    const/4 v7, 0x1
-
-    aput-object v6, v5, v7
-
-    invoke-virtual {v3, v0, v5}, Ljava/lang/Class;->getMethod(Ljava/lang/String;[Ljava/lang/Class;)Ljava/lang/reflect/Method;
-
-    move-result-object v3
-
-    const/4 v5, 0x0
-
-    new-array v4, v4, [Ljava/lang/Object;
-
-    aput-object p0, v4, v1
-
-    aput-object v2, v4, v7
-
-    invoke-virtual {v3, v5, v4}, Ljava/lang/reflect/Method;->invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v2
-
-    check-cast v2, Ljava/lang/Integer;
-
-    invoke-virtual {v2}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    if-eqz v2, :cond_0
-
-    invoke-direct {p0, v0, v2}, Lmiui/external/Application;->handleUnknownError(Ljava/lang/String;I)V
-    :try_end_0
-    .catchall {:try_start_0 .. :try_end_0} :catchall_0
-
-    return v1
-
-    :cond_0
-    return v7
-
-    :catchall_0
-    move-exception v0
+    .locals 1
 
-    invoke-direct {p0, v0}, Lmiui/external/Application;->handleGenericError(Ljava/lang/Throwable;)V
+    const/4 v0, 0x1
 
-    return v1
+    return v0
 .end method
 
 .method private loadSdk()Z
-    .locals 5
-
-    const-string v0, "com.miui.core"
-
-    const/4 v1, 0x0
-
-    :try_start_0
-    invoke-static {}, Lmiui/external/SdkHelper;->isMiuiSystem()Z
-
-    move-result v2
-
-    if-nez v2, :cond_0
-
-    const-string v2, "miui"
-
-    const/4 v3, 0x0
-
-    invoke-static {v3, v0, v2}, Lmiui/external/SdkHelper;->getApkPath(Landroid/content/Context;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v2
-
-    invoke-static {v3, v0}, Lmiui/external/SdkHelper;->getLibPath(Landroid/content/Context;Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v0
-
-    const-class v4, Lmiui/external/Application;
-
-    invoke-virtual {v4}, Ljava/lang/Class;->getClassLoader()Ljava/lang/ClassLoader;
-
-    move-result-object v4
-
-    invoke-static {v2, v3, v0, v4}, Lmiui/external/SdkLoader;->load(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/ClassLoader;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
-    sget-object v0, Lmiui/external/SdkConstants$SdkError;->NO_SDK:Lmiui/external/SdkConstants$SdkError;
-
-    invoke-static {v0}, Lmiui/external/SdkErrorInstrumentation;->handleSdkError(Lmiui/external/SdkConstants$SdkError;)V
-    :try_end_0
-    .catchall {:try_start_0 .. :try_end_0} :catchall_0
-
-    return v1
+    .locals 1
 
-    :cond_0
     const/4 v0, 0x1
 
     return v0
-
-    :catchall_0
-    move-exception v0
-
-    invoke-direct {p0, v0}, Lmiui/external/Application;->handleGenericError(Ljava/lang/Throwable;)V
-
-    return v1
 .end method
 
 .method private startSdk()Z
-    .locals 7
-
-    const-string v0, "start"
-
-    const/4 v1, 0x0
-
-    :try_start_0
-    new-instance v2, Ljava/util/HashMap;
-
-    invoke-direct {v2}, Ljava/util/HashMap;-><init>()V
-
-    invoke-static {}, Lmiui/external/SdkEntranceHelper;->getSdkEntrance()Ljava/lang/Class;
-
-    move-result-object v3
-
-    const/4 v4, 0x1
-
-    new-array v5, v4, [Ljava/lang/Class;
-
-    const-class v6, Ljava/util/Map;
-
-    aput-object v6, v5, v1
-
-    invoke-virtual {v3, v0, v5}, Ljava/lang/Class;->getMethod(Ljava/lang/String;[Ljava/lang/Class;)Ljava/lang/reflect/Method;
-
-    move-result-object v3
-
-    const/4 v5, 0x0
-
-    new-array v6, v4, [Ljava/lang/Object;
-
-    aput-object v2, v6, v1
-
-    invoke-virtual {v3, v5, v6}, Ljava/lang/reflect/Method;->invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v2
-
-    check-cast v2, Ljava/lang/Integer;
-
-    invoke-virtual {v2}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    if-ne v2, v4, :cond_0
-
-    sget-object v0, Lmiui/external/SdkConstants$SdkError;->LOW_SDK_VERSION:Lmiui/external/SdkConstants$SdkError;
-
-    invoke-static {v0}, Lmiui/external/SdkErrorInstrumentation;->handleSdkError(Lmiui/external/SdkConstants$SdkError;)V
-
-    return v1
-
-    :cond_0
-    if-eqz v2, :cond_1
-
-    invoke-direct {p0, v0, v2}, Lmiui/external/Application;->handleUnknownError(Ljava/lang/String;I)V
-    :try_end_0
-    .catchall {:try_start_0 .. :try_end_0} :catchall_0
-
-    return v1
-
-    :cond_1
-    return v4
-
-    :catchall_0
-    move-exception v0
+    .locals 1
 
-    invoke-direct {p0, v0}, Lmiui/external/Application;->handleGenericError(Ljava/lang/Throwable;)V
+    const/4 v0, 0x1
 
-    return v1
+    return v0
 .end method
 
 
-- 
2.43.0

